src/main/java/ch/admin/bar/siard2/jdbc/PostgresMetaColumns.java

```diff
@@ -203,19 +203,60 @@ private String getTypeName(String sTypeName, int iType)
           rs.close();
         }
         ***/
+        String sSchemaName = this.getString(2); // TABLE_SCHEM
+        String sTableName = this.getString(3); // TABLE_NAME
+        String sColumnName = this.getString(4); // COLUMN_NAME
+
         if (PostgresType.INTERVAL.getKeyword().equals(sTypeName)) 
         {
-          String sSchemaName = this.getString(2);
-          String sTableName = this.getString(3);
-          String sColumnName = this.getString(4);
           sTypeName = getIntervalTypeName(_conn, sSchemaName, sTableName, sColumnName, sTypeName);
         }
+
+        //add type length/precision
+        sTypeName = addTypeLengthAndPrecision(sTypeName, iType, getAttTypMod(_conn, sSchemaName, sTableName, sColumnName));
       }
       catch(ParseException pe) { throw new SQLException("Parsing of "+sTypeName+" failed ("+EU.getExceptionMessage(pe)+")!"); }
     }
     return sTypeName;
   } /* getTypeName */
-  
+
+  private String addTypeLengthAndPrecision(String sTypeName, int iType, int atttypmod) {
+    if (atttypmod < 0){
+      return sTypeName;
+    }
+
+    String sPrecisionType = "_" + atttypmod;
+
+    if(iType == Types.VARCHAR || iType == Types.CHAR){
```

Comment:
There is a missing space before the opening curly bracket. It's correct on line 235 though.
Make sure to use intellijs auto-format (ctrl+alt+l). 
Select lines you want to format if you don't want to change the whole file.